### DISCUSSION TITLE
Create a check run, details_url is not being set

### DISCUSSION THREAD

**Discussion by bkwhite** [community/community#26757] [Actions]
<p>Hello all!</p>
<p>I am using the JavaScript packages (<code>@actions/core</code> &amp; <code>@actions/github</code>) inside my custom GitHub action.   <a href="https://octokit.github.io/rest.js/v18#checks-create" rel="noopener nofollow ugc">According to the rest api</a> I should be able to provide a <code>details_url</code> as an option of my creation request.   Now matter what I seem to do the <code>details_url</code> does not seem to be set in the check that is displayed on the pull request page.  I think I am missing something here.</p>
<p>Thanks for your help!</p>
<p>Here is the repo: <a href="https://github.com/bkwhite/checks-playground" class="inline-onebox" rel="noopener nofollow ugc">GitHub - bkwhite/checks-playground</a></p>

**Comment by joshmgross** [comment]
<p>:wave: Hey <a class="mention" href="/bkwhite">@bkwhite</a>, that repository is private so I don’t have access to it.</p>
<p>Are you creating these check runs with the <code>GITHUB_TOKEN</code> or a token generated by a GitHub app?</p>

**Comment by bkwhite** [comment]
<p>Hello! Sorry that repo got moved to an org.</p>
<p>Here is the new link <a href="https://github.com/soomo/cypress-check/blob/4237cf576cfa4e7c10da9f81efaf163ce65ca9f3/src/index.ts#L65-L82" class="inline-onebox" rel="noopener nofollow ugc">cypress-check/index.ts at 4237cf576cfa4e7c10da9f81efaf163ce65ca9f3 · soomo/cypress-check · GitHub</a>.  We were trying to set the <code>details_url</code> in the <code>octokit.checks.create</code> function and not having any luck.</p>
<p>We are using the token that is passed from the action into the check run.</p>

**Comment by joshmgross** [comment, marked as answer]
<p>We don’t support showing a <code>details_url</code> for check runs created with the Actions <code>GITHUB_TOKEN</code>.</p>
<p>If you want this scenario to work, you’d need to use a different token from a GitHub App.</p>

**Comment by rabernat** [comment]
<p>Thanks to everyone helping here!</p>
<p><a class="mention" href="/joshmgross">@joshmgross</a> - can you give me some hints about how to “use a different token”? What kind of token would work here? Does it have to be an app, or can I use a personal access token?</p>

**Comment by joshmgross** [comment]
<p>It would need to be from a GitHub App, as the Checks API only works with Apps.</p>

**Comment by matthchr** [comment]
<p>Hi <a class="mention" href="/joshmgross">@joshmgross</a> thanks for answering this question. I just ran into this too.</p>
<p>Is there any documentation describing this limitation? Would you be willing to expand on why it’s not supported? It seems like it’d be useful for a bunch of cases. My case is doing something like <a href="https://github.com/imjohnbo/ok-to-test" class="inline-onebox" rel="noopener nofollow ugc">GitHub - imjohnbo/ok-to-test: Example workflow configuration showing how to use GitHub Actions secrets in pull requests from forks 🍴🔑</a> - except I’d like to fix up the action shown in the PR so that it links to the action triggered by the ok-to-test command so that logs are easier to find.</p>

**Comment by joshmgross** [comment]
<p>:wave: We have some documentation here: <a href="https://docs.github.com/en/rest/guides/getting-started-with-the-checks-api" class="inline-onebox">Getting started with the Checks API - GitHub Docs</a></p>
<p>The Checks API was built around GitHub Apps, so there’s a pretty tight coupling between Apps and Checks, more than just an authorization restriction.</p>

**Comment by FreundDK** [comment]
I also ran into this limitation, when trying to create a secure way of receiving pull requests, following this:
https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

My PR handler will store the files and the CI/CD will run on workflow_run.
If I don't register a check from the CI/CD workflow, the PR will look like everything has passed and the code (potentially malicious) can be approved. Checks works fine - but as other people have stated, details_url doesn't (apparently by design).

Too bad, but I will stop looking for solutions.

**Comment by jonathanwiesel** [comment]
Would be nice to have this limitation described in the REST API for Checks to avoid unnecessary head scratching
